gregBorchers/Homework2GregBorchers-cards-template1a.js

```diff
@@ -0,0 +1,169 @@
+                , "Four of Clubs"
+                                
+                , "Five of Hearts"
+                , "Five of Diamonds"
+                , "Five of Spades"
+                , "Five of Clubs"
+
+                , "Six of Hearts"
+                , "Six of Diamonds"
+                , "Six of Spades"
+                , "Six of Clubs"
+
+                , "Seven of Hearts"
+                , "Seven of Diamonds"
+                , "Seven of Spades"
+                , "Seven of Clubs"
+
+                , "Eight of Hearts"
+                , "Eight of Diamonds"
+                , "Eight of Spades"
+                , "Eight of Clubs"
+
+                , "Nine of Hearts"
+                , "Nine of Diamonds"
+                , "Nine of Spades"
+                , "Nine of Clubs"
+
+                , "Ten of Hearts"
+                , "Ten of Diamonds"
+                , "Ten of Spades"
+                , "Ten of Clubs"
+
+                , "Jack of Hearts"
+                , "Jack of Diamonds"
+                , "Jack of Spades"
+                , "Jack of Clubs"
+
+                , "Queen of Hearts"
+                , "Queen of Diamonds"
+                , "Queen of Spades"
+                , "Queen of Clubs"
+
+                , "King of Hearts"
+                , "King of Diamonds"
+                , "King of Spades"
+                , "King of Clubs" ];
+
+// **** alternative to build deckOfCards[] **** 
+// Loop across two string arrays to fill the arrary
+// Did not do this because having the static array was a useful visual reference
+// Here is code that would fill the array
+
+// var suitNames = ["Hearts", "Diamonds", "Spades", "Clubs"];
+// var rankNames = ["Ace","Two","Three","Four","Five","Six","Seven","Eight","Nine","Ten","Jack","Queen","King"];
+// var deckOfCards2 = [];
+// var cardID =0;
+
+// for (var i = 0; i < rankNames.length; i++) {
+//   for (var j = 0; j < suitNames.length; j++) {
+//     deckOfCards2[cardID] =  rankNames[i] + " of " + suitNames[j]
+//     console.log( deckOfCards2[cardID] );
+//     cardID++;
+//   };
+// };
+
+
+
+//****************************************** rank(card)
+// function rank(card) { // --> 1..13
+// }
+
+function rank(id) {
+  return Math.trunc(id/4)+1;
+}
+
+
+
+//****************************************** suit(card)
+// function suit(card) { // --> 1..4
+// }
+
+function suit(id) {
+  return (id%4)+1;
+}
+
+
+
+//****************************************** cardID(rank,suit) 
+// function cardID(rank,suit) { // --> 0..51
+// }
+
+function cardID(rank,suit) {
+  return ((rank - 1) * 4 + (suit - 1));
+}
+
+
+//****************************************** color(card)
+// function color(card) { // -->"red","black"
+// }
+
+var color = function(id){
+  var cardColor = "";
+  //TODO assert(id >= 0 && id <=3, "id is out of range");
+  if ( (suit(id) === 0) || (suit(id) === 1 ) )
+  {
+    cardColor = "red";
+  }
+  else
+  {
+    cardColor = "black";
+  }
+  return cardColor;
+};
```

Comment:
Remember that `suit()` returns a value from 1 to 4, not 0 to 3! Hearts are getting called black cards at the moment.